test/puppetlabs/ssl_utils/extensions_utils_test.clj

```diff
@@ -0,0 +1,23 @@
+(ns puppetlabs.ssl-utils.extensions-utils-test
+  (:require [clojure.test :refer :all]
+            [puppetlabs.ssl-utils.simple-test :as simple-test]
+            [puppetlabs.ssl-utils.core :as ssl-utils]
+            [puppetlabs.ssl-utils.simple :as simple]))
+
+(deftest general-names
+  (testing "Can encode and decode all General Names types"
+    (let [gns {:rfc822-name ["foo@bar.com"]
+               :dns-name ["localhost.localdomain"]
+               :uri ["file:///foo/bar"]
+               :ip ["192.168.69.90"]
+               :registered-id ["1.2.3.4"]}
```

Comment:
Yeah, should we be testing all of these: https://github.com/dankreek/jvm-ssl-utils/blob/SERVER-816/master/bug-decoding-general-names/src/java/com/puppetlabs/ssl_utils/ExtensionsUtils.java#L825-L833